core/vm/evm.go

```diff
@@ -209,30 +227,24 @@ func (self *EVM) Main(trx *Transaction, opts ExecutionOpts) (ret ExecutionResult
 		}
 		caller.SubBalance(gas_fee)
 	}
-
-	var err error
-	gas_intrinsic, err := IntrinsicGas(self.trx.Input, contract_creation, self.rules.IsHomestead)
-	if err != nil {
-		ret.ConsensusErr = util.ErrorString(err.Error())
-		return
-	}
-	if gas_left < gas_intrinsic {
-		ret.ConsensusErr = ErrIntrinsicGas
-		return
-	}
 	gas_left -= gas_intrinsic
 
 	if contract_creation {
 		ret.CodeRetval, ret.NewContractAddr, gas_left, err = self.create_1(caller, self.trx.Input, gas_left, self.trx.Value)
 	} else {
 		acc_to := self.state.GetAccount(self.trx.To)
-		caller.IncrementNonce()
+		// This is left here only because during dpos_test runs `self.trx.Nonce == nil`
+		if self.trx.Nonce == nil {
```

Comment:
trx.Nonce == nil means contract creation so we should rather fix dpos_test than doing this...